lighthouse-core/runner.js

```diff
@@ -25,6 +25,32 @@ const fs = require('fs');
 const path = require('path');
 const url = require('url');
 
+function runAudit(audit, artifacts) {
+  const status = `Evaluating: ${audit.meta.description}`;
+
+  return Promise.resolve().then(_ => {
+    log.log('status', status);
+
+    // Return an early error if an artifact required for the audit is missing.
+    for (const artifactName of audit.meta.requiredArtifacts) {
+      // TODO: need pass-specific check of networkRecords and traces
```

Comment:
> are there any examples here that used a trace from anything but `Audit.DEFAULT_PASS`?

not currently, but there are audits that use `networkRecords` not from `DEFAULT_PASS`.

we could do something hacky like `requiredArtifacts: [traces.dbw]` (or something) and get the pass name it will use from that, but that's doubling down on having hardcoded pass names in the audit (which means in the config file you have to name your passes based on a string in an audit, and you e.g. can't run two audits on the same trace if their audits use different pass names) rather than the real long term solution, which is likely something around declaratively specifying in the config which pass an audit should process.